flows/src/main/java/com/softwaremill/jox/flows/LinesImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.softwaremill.jox.flows;
+
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+class LinesImpl {
+
+    static <T> Flow<String> lines(Charset charset, Flow<T> parentFlow) {
```

Comment:
here we can require that the parent flow emits ByteChunks?

Reply:
We can, but we'll need to cast parentFlow to `Flow<ByteChunk`, and I'm not sure this will be safe (I'll verify it) 

+ right now we can support both byte[] and ByteChunk

Reply:
So if the Flow contains Integers we get class cast exception on method call. 

We cannot check it before due to type erasure (we don't know what's inside Flow).

Reply:
hm but can't we simply say that the parameter should be a `Flow<ByteChunk> parentFlow`, then this will be verified at compile-time?

Reply:
We can but then in `Flows#lines` we need to cast `this` to `(Flow<ByteChunk>)` , and because of type erasure, such cast will be succesful for both `Flow<Integer>` as well as `Flow<ByteChunk`

Reply:
Hm right, we'd have to move `Flow.lines` to a different object, e.g. `ByteFlows.lines(...)`, wdyt?

Reply:
So I've added subclass `ByteFlow` that can be created from flow containing `byte[]` or `ByteChunk`. 

I've also added few factory methods to `Flows` so it's possible to create ByteFlow directly. 

There is also a possibility to change type to `ByteFlow` by running `toByteFlow` without parameters or with proper mapping function